Mage.Sets/src/mage/cards/r/RielleTheEverwise.java

```diff
@@ -0,0 +1,128 @@
+package mage.cards.r;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import mage.MageInt;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.dynamicvalue.common.CardsInControllerGraveyardCount;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.continuous.SetPowerSourceEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.WatcherScope;
+import mage.constants.Zone;
+import mage.filter.common.FilterInstantOrSorceryCard;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.token.HumanSoldierToken;
+import mage.watchers.Watcher;
+import mage.watchers.common.CardsCycledOrDiscardedThisTurnWatcher;
+
+public class RielleTheEverwise extends CardImpl{
+
+	public RielleTheEverwise(UUID ownerId, CardSetInfo setInfo) {
+		super(ownerId, setInfo, new CardType[]{CardType.CREATURE},"{1}{U}{R}");
+		
+		this.addSuperType(SuperType.LEGENDARY);
+		
+		this.subtype.add(SubType.HUMAN);
+		this.subtype.add(SubType.WIZARD);
+		
+		this.power = new MageInt(0);
+		this.toughness = new MageInt(3);
+		
+		this.addAbility(new SimpleStaticAbility(Zone.ALL, new SetPowerSourceEffect(
+                new CardsInControllerGraveyardCount(new FilterInstantOrSorceryCard("instant and sorcery cards")), Duration.EndOfGame)));
+		
+		this.addAbility(new RielleTheEverwiseTriggeredAbility(ownerId));
+		
+	}
+	
+	public RielleTheEverwise(final RielleTheEverwise card) {
+        super(card);
+    }
+
+	@Override
+	public Card copy() {
+		return new RielleTheEverwise(this);
+	}
+	
+	class RielleTheEverwiseTriggeredAbility extends TriggeredAbilityImpl{
+
+		public RielleTheEverwiseTriggeredAbility(UUID playerId) {
+			super(Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(new RielleTheEverwiseWatcher().getTimesDiscarded(playerId)), false);
```

Comment:
this won't get the correct amount of draws, you need to modify the effect in `checkTrigger`